### PR TITLE
[SPARK-36166][TESTS] Support Scala 2.13 test in `dev/run-tests.py`

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

For Apache Spark 3.2, this PR aims to support Scala 2.13 test in `dev/run-tests.py` by adding `SCALA_PROFILE` and in `dev/run-tests-jenkins.py` by adding `AMPLAB_JENKINS_BUILD_SCALA_PROFILE`.

In addition, `test-dependencies.sh` is skipped for Scala 2.13 because we currently don't maintain the dependency manifests yet. This will be handled after Apache Spark 3.2.0 release.

### Why are the changes needed?

To test Scala 2.13 with `dev/run-tests.py`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual. The following is the result. Note that this PR aims to **run** Scala 2.13 tests instead of **passing** them. We will have daily GitHub Action job via #33358 and will fix UT failures if exists.
```
$ dev/change-scala-version.sh 2.13

$ SCALA_PROFILE=scala2.13 dev/run-tests.py
...
========================================================================
Running Scala style checks
========================================================================
[info] Checking Scala style using SBT with these profiles:  -Phadoop-3.2 -Phive-2.3 -Pscala-2.13 -Pkubernetes -Phadoop-cloud -Phive -Phive-thriftserver -Pyarn -Pmesos -Pdocker-integration-tests -Pkinesis-asl -Pspark-ganglia-lgpl
...
========================================================================
Building Spark
========================================================================
[info] Building Spark using SBT with these arguments:  -Phadoop-3.2 -Phive-2.3 -Pscala-2.13 -Pspark-ganglia-lgpl -Pmesos -Pyarn -Phive-thriftserver -Pkinesis-asl -Pkubernetes -Pdocker-integration-tests -Phive -Phadoop-cloud test:package streaming-kinesis-asl-assembly/assembly
...

[info] Building Spark assembly using SBT with these arguments:  -Phadoop-3.2 -Phive-2.3 -Pscala-2.13 -Pspark-ganglia-lgpl -Pmesos -Pyarn -Phive-thriftserver -Pkinesis-asl -Pkubernetes -Pdocker-integration-tests -Phive -Phadoop-cloud assembly/package
...

========================================================================
Running Java style checks
========================================================================
[info] Checking Java style using SBT with these profiles:  -Phadoop-3.2 -Phive-2.3 -Pscala-2.13 -Pspark-ganglia-lgpl -Pmesos -Pyarn -Phive-thriftserver -Pkinesis-asl -Pkubernetes -Pdocker-integration-tests -Phive -Phadoop-cloud
...

========================================================================
Building Unidoc API Documentation
========================================================================
[info] Building Spark unidoc using SBT with these arguments:  -Phadoop-3.2 -Phive-2.3 -Pscala-2.13 -Pspark-ganglia-lgpl -Pmesos -Pyarn -Phive-thriftserver -Pkinesis-asl -Pkubernetes -Pdocker-integration-tests -Phive -Phadoop-cloud unidoc
...

========================================================================
Running Spark unit tests
========================================================================
[info] Running Spark tests using SBT with these arguments:  -Phadoop-3.2 -Phive-2.3 -Pscala-2.13 -Pspark-ganglia-lgpl -Pmesos -Pyarn -Phive-thriftserver -Pkinesis-asl -Pkubernetes -Pdocker-integration-tests -Phive -Phadoop-cloud test
...
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @gengliangwang , @HyukjinKwon , @sarutak

File: dev/run-tests.py
Comment (HyukjinKwon) [inline dev/run-tests.py:650]: @dongjoon-hyun, since we're touching here shall we name and add `AMPLAP_JENKINS_BUILD_SCALA_PROFILE` at https://github.com/apache/spark/blob/master/dev/run-tests-jenkins.py#L177-L179 too just for consistency? Since the change is logically very simple, I think we won't bother have to test.

**Review by HyukjinKwon — APPROVED** [review]
Looks good

File: dev/run-tests.py
Comment (HyukjinKwon) [reply dev/run-tests.py:650]: (I was just double checking a similar commit like https://github.com/apache/spark/commit/4a73bed3180aeb79c92bb19aea2ac5a97899731a)

**Review by sarutak — APPROVED** [review]
Thanks @dongjoon-hyun ! LGTM.

File: dev/run-tests.py
Comment (dongjoon-hyun) [reply dev/run-tests.py:650]: Thanks! Got it.

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by dongjoon-hyun** [comment]
GitHub Action is tested already and the last commit affects Jenkins only. After checking Jenkins trigger, I'll merge this PR. Thank you, @HyukjinKwon and @sarutak .

**Comment by dongjoon-hyun** [comment]
Hmm. Jenkins PR Builder seems to be stuck.The last run was 30 minutes ago.
- https://amplab.cs.berkeley.edu/jenkins/view/Pull%20Request%20Builders/job/SparkPullRequestBuilder/

**Comment by dongjoon-hyun** [comment]
I'll merge this and test later when it comes back. If there is a missing part for Jenkins, I'll make a follow-up. 
For me, Jenkins is not a thing we can access or invest more.

**Comment by HyukjinKwon** [comment]
I think that's fine to go ahead and merge.

**Comment by gengliangwang** [comment]
Thanks @dongjoon-hyun ! Late LGTM.

**Pull request closed without merging**
